app/services/data_stage/process_school_changes.rb

```diff
@@ -0,0 +1,90 @@
+# frozen_string_literal: true
+
+module DataStage
+  class ProcessSchoolChanges < ::BaseService
+    include GiasTypes
+
+    UNHANDLED_ATTRIBUTES = %w[
+      administrative_district_code
+      administrative_district_name
+      school_type_code
+      school_type_name
+      section_41_approved
+      ukprn
+    ].freeze
+
+    def call
+      DataStage::SchoolChange.unhandled.status_changed.find_each do |change|
+        next if has_unhandled_attributes?(change.attribute_changes)
+
+        if change.attribute_changes.key? "school_status_code"
+          handle_status_code_change(change)
+        end
+      end
+    end
+
+  private
+
+    def has_unhandled_attributes?(attributes)
+      (UNHANDLED_ATTRIBUTES & attributes.keys).any?
+    end
+
+    def handle_status_code_change(change)
+      from_code = change.attribute_changes["school_status_code"]&.first
+      to_code = change.attribute_changes["school_status_code"]&.last
+
+      if open_status_code?(from_code) && !open_status_code?(to_code)
+        close_school(change)
+      elsif !open_status_code?(from_code) && open_status_code?(to_code)
+        open_school(change)
+      else
+        # just sync then
+        change.school.create_or_sync_counterpart!
+      end
+    end
+
+    def open_school(change)
+      school = change.school
+
+      # is there a predecessor
+      last_link = school.school_links.order(created_at: :asc).last
+
+      if last_link.nil? || last_link.link_type == "Predecessor"
+        DataStage::School.transaction do
+          # no predecessor or simple predecessor - add new school
+          school.create_or_sync_counterpart!
+
+          # transfer assets if predecessor
+          if last_link.present?
+            predecessor = ::School.find_by(urn: last_link.link_urn)
+            transfer_assets_from_predecessor(school: school.counterpart, predecessor: predecessor) if predecessor.present?
+          end
+
+          change.update!(handled: true)
+        end
+      end
+    end
+
+    def close_school(change)
+      # let any related school open transfer any assets
+      # just sync attribute changes
+      DataStage::School.transaction do
+        # only sync changes don't create
+        change.school.create_or_sync_counterpart! if change.school.counterpart.present?
+        change.update!(handled: true)
+      end
+    end
```

Comment:
What does `let any related school open transfer any assets` mean?

Also, why do we check if the counterpart is present here? `create_or_sync_counterpart` does it for us (I think)

Reply:
Yes missing a word/grammar in the comment, it should be "let any related 'school open' change handle the transfer of assets"
In this case we don't want to create a counterpart if it doesn't exist as we're closing the school - the `create_or_sync_counterpart!` will created it if it doesn't exist.

Reply:
Moving this class to a separate PR